Assets/Packages/com.studio23.ss2.splashscreensystem/Runtime/Core/SplashScreenBehaviour.cs

```diff
@@ -0,0 +1,52 @@
+using Cysharp.Threading.Tasks;
+using Studio23.SS2.SplashScreenSystem.Data;
+using Studio23.SS2.SplashScreenSystem.UI;
+using System;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace Studio23.SS2.SplashScreenSystem.Core
+{
+    [System.Serializable]
+    public class UnityEventWithBool : UnityEvent<bool> {}
+    public class SplashScreenBehaviour : MonoBehaviour
+    {
+        public SplashScreen[] SplashScreens;
+        public UnityEventWithBool OnEULAResponse;
+        public UnityEvent OnSplashScreenCompleted;
+
+        private bool _eulaButtonClicked;
+
+        private void Start()
+        {
+            ShowSplashScreen();
+        }
+
+        private async void ShowSplashScreen()
+        {
+            foreach (var s in SplashScreens)
```

Comment:
Use a bit more concise naming rather than singular letters. In this case, instead of `var s in SplashScreens` it can be 
`var splash in SplashScreens`